### DISCUSSION TITLE
How to insert a dictionary element in a ordered_json dict

### DISCUSSION THREAD

**Discussion by SrideviRamesh** [nlohmann/json#2390] [Q&A] [answered]
I want to insert elements from dict 1 and dict 2 in to dict 3.. I expect output dictionary dict3 not to be sorted in alphabetic order so am using ordered_json

```cpp
#include <nlohmann/json.hpp>
#include  <iostream>

using ordered_json = nlohmann::ordered_json;
int main()
{
    ordered_json dict1;
    dict1["name"] = "sridevi";
    dict1["age"]= "28";
    ordered_json dict2;
    dict2["place"] = "bangalore";
    dict2["hobby"] = "reading";
    ordered_json dict3;
    dict3["number"] = "30";
    dict3.insert(dict1.begin(),dict1.end());
    dict3.insert(dict2.begin(),dict2.end());
    std::cout << dict3.dump(4) << std::endl;
}
```
when i compile this code i see below error

nlohmann/json.hpp: In instantiation of ‘void nlohmann::basic_json<ObjectType, ArrayType, StringType, BooleanType, NumberIntegerType, NumberUnsignedType, NumberFloatType, AllocatorType, JSONSerializer, BinaryType>::insert(nlohmann::basic_json<ObjectType, ArrayType, StringType, BooleanType, NumberIntegerType, NumberUnsignedType, NumberFloatType, AllocatorType, JSONSerializer, BinaryType>::const_iterator, nlohmann::basic_json<ObjectType, ArrayType, StringType, BooleanType, NumberIntegerType, NumberUnsignedType, NumberFloatType, AllocatorType, JSONSerializer, BinaryType>::const_iterator) [with ObjectType = nlohmann::ordered_map; ArrayType = std::vector; StringType = std::__cxx11::basic_string<char>; BooleanType = bool; NumberIntegerType = long long int; NumberUnsignedType = long long unsigned int; NumberFloatType = double; AllocatorType = std::allocator; JSONSerializer = nlohmann::adl_serializer; BinaryType = std::vector<unsigned char>; nlohmann::basic_json<ObjectType, ArrayType, StringType, BooleanType, NumberIntegerType, NumberUnsignedType, NumberFloatType, AllocatorType, JSONSerializer, BinaryType>::const_iterator = nlohmann::detail::iter_impl<const nlohmann::basic_json<nlohmann::ordered_map> >]’:

please suggest the fix... Thanks

**Comment by nlohmann** [comment, marked as answer]
Do you use the most recent version from the develop branch?

**Reply by SrideviRamesh** [reply]
I am using version 3.9.0

**Reply by nlohmann** [reply]
Can you please try with 3.9.1 - there was a fix on ordered_json.

**Reply by SrideviRamesh** [reply]
Sure i will try and get back

**Reply by nlohmann** [reply]
And?
